src/main/java/dao/ReaderServiceImp.java

```diff
@@ -0,0 +1,26 @@
+package dao;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import validator.Validator;
+import validator.ValidatorImp;
+
+public class ReaderServiceImp implements ReaderService {
+    private Validator validator = new ValidatorImp();
+
+    @Override
+    public List<String> readFromFile(String filePath) {

```

Comment:
It's not obvious that reader service has inner validation. In this case if we would like to reuse this class in future this may produce some problems. Let's move validation to the upper level (e.g ReportService)